partition/roles/sonic/tasks/generate_config_db.yaml

```diff
@@ -0,0 +1,45 @@
+---
+- name: Create temporary directory
+  file:
+    path: /tmp/sonic
+    state: directory
+
+- name: Render input values template
+  template:
+    src: sonic-config.yaml.j2
+    dest: /tmp/sonic/sonic-config.yaml
+
+- name: Copy current config_db.json into temporary directory
+  copy:
+    src: /etc/sonic/config_db.json
+    dest: /tmp/sonic/config_db.json
+    mode: 0644
+    remote_src: true
+
+- name: Run sonic-configdb-utils
+  community.docker.docker_container:
+    name: sonic-configdb-utils
+    image: "{{ sonic_configdb_utils_image_name }}:{{ sonic_configdb_utils_image_tag }}"
+    pull: true
+    detach: false
+    cleanup: true
+    command: generate
+    volumes:
+      - /usr/share/sonic/device:/usr/share/sonic/device:ro
```

Comment:
Why you mount the whole device tree instead of only the directories `hwsku` and `platform`?
```
$ ls -l /usr/share/sonic/
total 16
drwxr-xr-x 1 root root 4096 Feb 13 10:11 device
lrwxrwxrwx 1 root root   69 Feb 13 10:11 hwsku -> /usr/share/sonic/device/x86_64-accton_as7726_32x-r0/Accton-AS7726-32X
lrwxrwxrwx 1 root root   51 Feb 13 10:11 platform -> /usr/share/sonic/device/x86_64-accton_as7726_32x-r0
drwxr-xr-x 2 root root 4096 Feb 13 10:11 scripts
drwxr-xr-x 2 root root  710 Jun 19  2024 templates
```

Reply:
Depending on the SONiC version the `platform` directory might be empty.